__tests__/app.test.js

```diff
@@ -270,28 +272,150 @@ describe("POST /api/articles/:article_id/comments", () => {
-    
+  test("400: Should return 400 if the body is missing from body request", async () => {
+    const response = await request(app).post("/api/articles/9/comments").send();
+
     expect(response.status).toBe(400);
-    expect(response.body.msg).toBe('Request body is missing');
+    expect(response.body.msg).toBe("Request body is missing");
   });
-  test('400: Should return 400 if the author is missing from body request', async () => {
+  test("400: Should return 400 if the author is missing from body request", async () => {
     const response = await request(app)
-      .post('/api/articles/9/comments')
-      .send([{ body: 'Amazing!' }]); 
-    
+      .post("/api/articles/9/comments")
+      .send([{ body: "Amazing!" }]);
+
     expect(response.status).toBe(400);
-    expect(response.body.msg).toBe('Author is required');
+    expect(response.body.msg).toBe("Author is required");
   });
-  test('400: Should return 400 if the body is missing from body request', async () => {
+  test("400: Should return 400 if the body is missing from body request", async () => {
     const response = await request(app)
-      .post('/api/articles/9/comments')
-      .send([{ author: 'butter_bridge' }]); 
-    
+      .post("/api/articles/9/comments")
+      .send([{ author: "butter_bridge" }]);
+
     expect(response.status).toBe(400);
-    expect(response.body.msg).toBe('body text is required');
+    expect(response.body.msg).toBe("body text is required");
+  });
+});
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+describe("GET /api/articles (topic query)", () => {
+  test("Should return articles filtered by topic when topic query is provided", async () => {
+    const response = await request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200);
+    const {
+      body: { articles },
+    } = response;
+    const expected = {
+     article_id: 3,
+     article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+     author: "icellusedkars",
+     comment_count: "2",
+     created_at: "2020-11-03T09:12:00.000Z",
+     title: "Eight pug gifs that remind me of mitch",
+     topic: "mitch",
+     votes: 0,
+    };
+
+    expect(articles[0]).toMatchObject(expected);
+    expect(articles).toHaveLength(12);
+  });
+
+  test("404: Should return 404 error for invalid topic", async () => {
+    const response = await request(app)
+      .get("/api/articles?topic=HELLO")
+      .expect(404);
+    const {
+      body: { msg },
+    } = response;
+    expect(msg).toBe("Topic does not exist");
+  });
+
+  test("Should return all articles if passed an empty topic query ", async () => {
+    const response = await request(app).get("/api/articles?topic=").expect(200);
+    const {
+      body: { articles },
+    } = response;
+    expect(articles).toHaveLength(13);
```

Comment:
There is a more nuanced case we can test for: 

 - a topic that exists but there are no articles for it